### PR TITLE
add mysql like 'show tables' statement to list all available tables

### PR DESCRIPTION
**Pull request by jossephus**
image

### CODE REVIEW

**Comment by AmrDeveloper** [comment]
Hello @jossephus,

Thank you for implementing this cool feature, can you please fix clippy comments on this PR then we need to fix it too on other PR and handle conflict

Thank you

File: crates/gitql-engine/src/engine.rs [hunk 1]
```diff
@@ -178,3 +180,26 @@ fn apply_distinct_on_objects_group(gitql_object: &mut GitQLObject, hidden_select
         gitql_object.groups[0].rows.append(&mut new_objects.rows);
     }
 }
+
+pub fn evaluate_show_tables_query(
+    env: &mut Environment,
+    _: &Box<dyn DataProvider>,
```
Comment (AmrDeveloper) [inline crates/gitql-engine/src/engine.rs:186]: If you don't need the DataProvider you can remove it from Parameters

File: crates/gitql-engine/src/engine.rs [hunk 2]
```diff
@@ -178,3 +180,26 @@ fn apply_distinct_on_objects_group(gitql_object: &mut GitQLObject, hidden_select
...
+
+    for table in env.schema.tables_fields_names.keys() {
+        gitql_object.groups.push(Group {
+            rows: vec![Row {
+                values: vec![Value::Text(table.to_owned().to_owned())],
+            }],
+        })
+    }
+
+    return Ok(EvaluationResult::SelectedGroups(
```
Comment (AmrDeveloper) [inline crates/gitql-engine/src/engine.rs:201]: You can remove `return` and `;` and keep the expression

**Review by AmrDeveloper — CHANGES_REQUESTED** [review]

**Comment by jossephus** [comment]
Sure. I didnt run clippy. I am fixing them now. Thanks.

**Comment by jossephus** [comment]
Clippy suggestions are implemented now.

**Comment by AmrDeveloper** [comment]
> Clippy suggestions are implemented now.

Thank you and welcome to the project

**Pull request merged**